Feat: Consume new OAS Isomorphic SDK (@W-19105545@)

adamraya opened the pull request:
<!--- Provide a short summary of your changes in the Title field above -->

# Description

<!--- A longer summary of your changes, including: a description of the issue that you’re addressing, a list of required dependencies (if applicable), and any other relevant context. -->

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [ ] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1)

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [ ] Changes are covered by test cases
- [ ] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

joeluong-sfcc committed:
refactor

joeluong-sfcc committed:
fix build issues

joeluong-sfcc committed:
trim uncessarry changes

joeluong-sfcc committed:
remove most usages of

joeluong-sfcc committed:
fix tests

joeluong-sfcc committed:
update test files

joeluong-sfcc committed:
fix unit tests

adamraya committed:
Bump commerce-sdk-isomorphic

adamraya committed:
Bump commerce-sdk-isomorphic

cc-prodsec commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/21feb81f-98ce-46b6-9926-85712462c7cf?refs=6583c28c5300d4259fe849f9474e079b4b4d4c64&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/cc-mobify-sdk/pr-checks/21feb81f-98ce-46b6-9926-85712462c7cf/license?refs=6583c28c5300d4259fe849f9474e079b4b4d4c64&source=prChecksComment)

adamraya committed:
Merge branch 'develop' into adamraya/consume-oas-sdk

adamraya committed:
Fix ts error

adamraya committed:
Use caret for the commerce-sdk-isomorphic version

adamraya committed:
Merge branch 'develop' into adamraya/consume-oas-sdk

adamraya committed:
Add CI logs

adamraya committed:
fix: rename ShopperSeo directory to ShopperSEO for case-sensitive filesystems

This fixes CI failures on Linux where the case-sensitive filesystem
couldn't find ShopperSEO files that were committed as ShopperSeo.

The issue occurred because:
- macOS (local): case-insensitive filesystem, both ShopperSeo/ShopperSEO work
- Linux (CI): case-sensitive filesystem, ShopperSeo ≠ ShopperSEO
- Code imports: expect ShopperSEO (uppercase O)
- Git had: ShopperSeo (lowercase o)

This ensures consistency across all platforms and matches the
commerce-sdk-isomorphic@4.0.0 naming convention.

adamraya committed:
Fix test-commerce-sdk-react

adamraya commented:
Closing in favour of original PR https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2879
